in_toto/models/layout.py

```diff
@@ -120,15 +99,21 @@ def read_from_file(filename='root.layout'):
   @staticmethod
   def read(data):
     """Static method to instantiate a new Layout from a Python dictionary """
+    signatures = data.get('signatures', [])
     steps = []
-    for step_data in data.get("steps"):
+
+    if 'signed' not in data:
+        data['signed'] = _Layout()
+
+    for step_data in data['signed'].get("steps"):
+      # FIXME: is this really needed?
```

Comment:
I think so yes, since we don't want to assign a list of step dictionaries to the `steps` attribute but a list of `Step` objects.